### PR TITLE
Refactor update for rooms

### PR DESCRIPTION
**Pull request by nirm123**
Changes to update and player class to use world and room.

### CODE REVIEW

File: assets/room.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Room:
-	def __init__(self, items = [], name = "", description = ""):
+	def __init__(self, x = 0, y = 0, name = "", items = [], description = ""):
 		self.items = items
-		self.name = name
+		self.name = " (" + repr(x) + ", " + repr(y) + ")"
```
Comment (Lathie) [inline assets/room.py]: Why repr instead of str? Just curious

File: update.py [hunk 2]
```diff
@@ -18,20 +18,32 @@ def update(self, pla, out):
 			def err():
 				mess.code = 0
 			def North():
-				mess.code = 1
-				pla.y_location += 1
+				if pla.room.y == 0:
```
Comment (Lathie) [inline update.py]: img-2018-03-18-011220

It looks like going south is going north? Logically, the coordinate system should be up/down flipped probably

Comment (Lathie) [reply update.py]: If there's a reason for the coordinate system to be this way let me know

File: assets/room.py (hunk 1 above)
Comment (nirm123) [reply assets/room.py]: There is no reason. I think the first time I needed to cast an int to a string, repr came up when I googled it.

File: update.py (hunk 2 above)
Comment (nirm123) [reply update.py]: I think in my head I was imagining the array as some sort of grid, which doesn't really make any sense anyway.

**Comment by nirm123** [comment]
I thought that was what I was doing but I might be wrong. The if statement is supposed to check if the player is at the edge of the array (y = 0 or y = len(world.grid[0]) - 1 if moving north or south). If the player is not in one of these rows, it is moved in the else statement. Am I misunderstanding something?

**Comment by Lathie** [comment]
Ah you're totally right, I'm not sure what I was looking at

**Pull request merged**